### PR TITLE
Reintroduces auto red alert when xenos are aboard a dropship

### PR DESCRIPTION
**Pull request by morrowwolf**
# About the pull request

This PR reintroduces auto red alert when xenos are aboard a dropship.

# Explain why it's good for the game

Oversight from shuttle port.

# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog

:cl: Morrow
add: Reintroduced auto red alert when xenos are aboard a dropship
/:cl:

### CODE REVIEW

**Review by harryob — APPROVED** [review]

File: code/modules/shuttle/shuttles/dropship.dm [hunk 1]
```diff
@@ -81,6 +81,16 @@
 	. = ..()
 	control_doors("force-lock-launch", "all", force=TRUE, asynchronous = FALSE)
 
+	for(var/area/checked_area in shuttle_areas)
+		for(var/turf/checked_turf in checked_area)
```
Comment (fira) [inline code/modules/shuttle/shuttles/dropship.dm:85]: IIRC you don't need to check turfs, if mobs are on map they're also in area

File: code/modules/shuttle/shuttles/dropship.dm [hunk 2]
```diff
@@ -81,6 +81,16 @@
...
+			for(var/mob/living/carbon/xenomorph/checked_xeno in checked_turf)
+				if(checked_xeno.stat == CONSCIOUS)
```
Comment (fira) [inline code/modules/shuttle/shuttles/dropship.dm:87]: Why not for inconscious?

File: code/modules/shuttle/shuttles/dropship.dm (hunk 1 above)
Comment (harryob) [reply code/modules/shuttle/shuttles/dropship.dm:85]: > The contents of areas are a little different. The turfs contained in the area are in the list along with any objs or mobs directly contained by those turfs.

according to the ref, yep!

File: code/modules/shuttle/shuttles/dropship.dm [hunk 3]
```diff
@@ -81,6 +81,16 @@
...
+					var/name = "Unidentified Lifesigns"
+					var/input = "Unidentified lifesigns detected onboard. Recommendation: lockdown of exterior access ports, including ducting and ventilation."
+					shipwide_ai_announcement(input, name, 'sound/AI/unidentified_lifesigns.ogg')
+					set_security_level(SEC_LEVEL_RED)
+					return
```
Comment (harryob) [inline code/modules/shuttle/shuttles/dropship.dm:92]:
```suggestion
			for(var/mob/living/carbon/xenomorph/checked_xeno in checked_turf)
				if(checked_xeno.stat == DEAD)
					continue
					
				var/name = "Unidentified Lifesigns"
				var/input = "Unidentified lifesigns detected onboard. Recommendation: lockdown of exterior access ports, including ducting and ventilation."
				shipwide_ai_announcement(input, name, 'sound/AI/unidentified_lifesigns.ogg')
				set_security_level(SEC_LEVEL_RED)
				return
```

**Review by harryob — APPROVED** [review]

**Pull request merged**